Add basic documentation (#21)
aeternity/installer: README.md

> [releases]: https://github.com/aeternity/aeternity/releases

Reviewer: These references usually stay at the bottom of the file

Author: 44f4c4d

* macOS High Sierra 10.13 (x86-64)
* macOS Mojave 10.14 (x86-64)
* macOS Catalina 10.15 (x86-64)
* Windows 10 (x86-64)

## Install dependencies

Package dependencies are:

* [Libsodium](https://download.libsodium.org/doc/)
* [Openssl](https://www.openssl.org)

### Ubuntu package

The package requires a libsodium v1.0.16 as `libsodium.so.23` shared object/library.

#### Ubuntu 18.04

Ubuntu 18.04 ships with libsodium 1.0.16, thus it can be installed with `apt` package manager:

```bash
sudo apt-get install libsodium23
```

The Ubuntu release binaries are built with `libssl1.0.0` (default Ubuntu 18.04 version is 1.1) requirement that can be installed with:

```bash
sudo apt-get install libssl1.0.0
```

#### Ubuntu 16.04

As Ubuntu 16.04 ships with older libsodium version than required, it must be installed from source.
A C compiler and related tools must be installed beforehand by running:

```bash
sudo apt-get install build-essential
```

then the library:

```bash
curl -O https://download.libsodium.org/libsodium/releases/libsodium-1.0.16.tar.gz
tar -xf libsodium-1.0.16.tar.gz && cd libsodium-1.0.16
./configure && make && sudo make install && sudo ldconfig
```

### macOS package

> Easiest way to install dependencies is using [Homebrew](https://brew.sh/):

Reviewer: You can add https://brew.sh/ in references as it is repeated at multiple places

Author: 44f4c4d

bash <(curl -s https://install.aeternity.io/install.sh)
```

> The latest argument is interpreted as version.

Reviewer: ```suggestion
The last argument is interpreted as a version.
```

Author: 9f52faf

The releases are published on [GitHub][releases].

To install an older specific version use:
```bash
bash <(curl -s https://install.aeternity.io/install.sh) 4.2.1
```

##### Additional options

- `--no-prompt` - Disable confirmation prompt

> - `--delete` - Deleting the directory where aeternity node will be installed

Reviewer: ```suggestion
- `--delete` - Clean the target directory before installation
```

Author: 9f52faf

Example:

```bash
bash <(curl -s https://install.aeternity.io/install.sh) --no-prompt [version]
```

##### Additional parameters

- `TARGET_DIR` - To override the path where aeternity node will be installed (default: `$HOME/aeternity/node`)

Example:

```bash
TARGET_DIR=/some/dir bash <(curl -s https://install.aeternity.io/install.sh)
```

## Supported platforms

* Ubuntu 16.04.3 LTS (x86-64)
* Ubuntu 18.04 LTS (x86-64)
* macOS High Sierra 10.13 (x86-64)
* macOS Mojave 10.14 (x86-64)
* macOS Catalina 10.15 (x86-64)

## Dependencies that will be installed automatically

* [Libsodium](https://download.libsodium.org/doc/)
* [Openssl](https://www.openssl.org)

> * gmp

Reviewer 2: [GMP lib](https://gmplib.org)

Author: 9f52faf

> See the documentation [here](https://github.com/aeternity/aeternity/blob/master/README.md) detailed instructions for manual installation.

Reviewer: This sentence seem is not grammatically correct.

Reviewer: `For manual installation see the [detailed instructions](https://github.com/aeternity/aeternity/blob/master/README.md).`

Author: a7b4ad4

# Run install tests

> Downloading the [Git repository](https://github.com/aeternity/installer) itself and:

Reviewer 2: I don't think this is necessary as it implies the developer is already working with this repo source.
I mean, this line is for developers of the tool, not users.

Author: 9f52faf

> Downloading the [Git repository](https://github.com/aeternity/installer) itself and:

Reviewer: ```suggestion
Clone the [git repository](https://github.com/aeternity/installer) and run:
```
